## Heart analysis using Numpy

In [3]:
import numpy as np

## Loding Dataset

In [8]:
data=np.genfromtxt("heart.csv", delimiter=",", skip_header=True)

## extracting relavent columns

#### FBS: Fasting blood sugar
#### ECG: resting electrocardiogram
#### CP: chest pain
#### Chol: colesterol
#### trestbps: resting blood pressure
#### Oldpeak: ST depression caused by activitiy in comparision to rest
#### Exang: excercise induced angina
#### Thalach: maximum heart rate achived

In [13]:
age=data[:, 0]
print (f"sex: {sex[:5]}")
target=data[:, -1]
print(f"target: {target [:5]}")
chol=data[:,4]
print(f"chol: {chol [:5]}")
sex=data [:,1]
print(f"age: {age[:5]}")
cp=data[:,2]
print(f"cp: {cp[:5]}")
thalach=data[:,7]
print(f"thalach: {thalach[:5]}")

sex: [1. 1. 0. 1. 0.]
target: [1. 1. 1. 1. 1.]
chol: [233. 250. 204. 236. 354.]
age: [63. 37. 41. 56. 57.]
cp: [3. 2. 1. 1. 0.]
thalach: [150. 187. 172. 178. 163.]


### Basic Descriptive statistics
* Mean, Median, SD, Min, Max
* sex and target excluded do to being binary values

In [14]:
print("---Basic Descriptive Statistics on chol  ---")
print(f"Mean: {round(np.mean(chol))}")
print(f"Median: {np.median(chol)}")
print(f"Standard Deviation: {round(np.std(chol))}")
print(f"Min: {np.min(chol)}")
print(f"Max: {np.max(chol)}")

---Basic Descriptive Statistics on chol  ---
Mean: 246
Median: 240.0
Standard Deviation: 52
Min: 126.0
Max: 564.0


In [15]:
print("---Basic Descriptive Statistics on age  ---")
print(f"Mean: {round(np.mean(age))}")
print(f"Median: {np.median(age)}")
print(f"Standard Deviation: {round(np.std(age))}")
print(f"Min: {np.min(age)}")
print(f"Max: {np.max(age)}")

---Basic Descriptive Statistics on age  ---
Mean: 54
Median: 55.0
Standard Deviation: 9
Min: 29.0
Max: 77.0


In [16]:
print("---Basic Descriptive Statistics on cp  ---")
print(f"Mean: {round(np.mean(cp))}")
print(f"Median: {np.median(cp)}")
print(f"Standard Deviation: {round(np.std(cp))}")
print(f"Min: {np.min(cp)}")
print(f"Max: {np.max(cp)}")

---Basic Descriptive Statistics on cp  ---
Mean: 1
Median: 1.0
Standard Deviation: 1
Min: 0.0
Max: 3.0


In [17]:
print("---Basic Descriptive Statistics on thalach  ---")
print(f"Mean: {round(np.mean(thalach))}")
print(f"Median: {np.median(thalach)}")
print(f"Standard Deviation: {round(np.std(thalach))}")
print(f"Min: {np.min(thalach)}")
print(f"Max: {np.max(thalach)}")

---Basic Descriptive Statistics on thalach  ---
Mean: 150
Median: 153.0
Standard Deviation: 23
Min: 71.0
Max: 202.0


## data filtering

In [18]:
print(f"Number of rows before filtering: {len(data)}")

Number of rows before filtering: 303


In [19]:
# how many are over 50 and have high colesterol
filtered_data=data[(age>50) & (chol>240)]
print(f"Number of rows after filtering: {len(filtered_data)}")

Number of rows after filtering: 111


## how many entries where:
* age ≥50,
* chol≥240,
* cp ≥2

In [23]:
filtered_data1=data[(age>=50) & (chol>=240)&(cp>=2)]
print(f"Number of rows after filtering: {len(filtered_data1)}")

Number of rows after filtering: 37


### Number of unique categories
* Unique can work for discreet variables

In [24]:
print(f"Unique bedroom counts: {np.unique(chol)}")
print(f" How many unique bedrooms: {len(np.unique(cp))}")

Unique bedroom counts: [126. 131. 141. 149. 157. 160. 164. 166. 167. 168. 169. 172. 174. 175.
 176. 177. 178. 180. 182. 183. 184. 185. 186. 187. 188. 192. 193. 195.
 196. 197. 198. 199. 200. 201. 203. 204. 205. 206. 207. 208. 209. 210.
 211. 212. 213. 214. 215. 216. 217. 218. 219. 220. 221. 222. 223. 224.
 225. 226. 227. 228. 229. 230. 231. 232. 233. 234. 235. 236. 237. 239.
 240. 241. 242. 243. 244. 245. 246. 247. 248. 249. 250. 252. 253. 254.
 255. 256. 257. 258. 259. 260. 261. 262. 263. 264. 265. 266. 267. 268.
 269. 270. 271. 273. 274. 275. 276. 277. 278. 281. 282. 283. 284. 286.
 288. 289. 290. 293. 294. 295. 298. 299. 300. 302. 303. 304. 305. 306.
 307. 308. 309. 311. 313. 315. 318. 319. 321. 322. 325. 326. 327. 330.
 335. 340. 341. 342. 353. 354. 360. 394. 407. 409. 417. 564.]
 How many unique bedrooms: 4


In [25]:
print(f"Unique bedroom counts: {np.unique(chol)}")
print(f" How many unique bedrooms: {len(np.unique(chol))}")

Unique bedroom counts: [126. 131. 141. 149. 157. 160. 164. 166. 167. 168. 169. 172. 174. 175.
 176. 177. 178. 180. 182. 183. 184. 185. 186. 187. 188. 192. 193. 195.
 196. 197. 198. 199. 200. 201. 203. 204. 205. 206. 207. 208. 209. 210.
 211. 212. 213. 214. 215. 216. 217. 218. 219. 220. 221. 222. 223. 224.
 225. 226. 227. 228. 229. 230. 231. 232. 233. 234. 235. 236. 237. 239.
 240. 241. 242. 243. 244. 245. 246. 247. 248. 249. 250. 252. 253. 254.
 255. 256. 257. 258. 259. 260. 261. 262. 263. 264. 265. 266. 267. 268.
 269. 270. 271. 273. 274. 275. 276. 277. 278. 281. 282. 283. 284. 286.
 288. 289. 290. 293. 294. 295. 298. 299. 300. 302. 303. 304. 305. 306.
 307. 308. 309. 311. 313. 315. 318. 319. 321. 322. 325. 326. 327. 330.
 335. 340. 341. 342. 353. 354. 360. 394. 407. 409. 417. 564.]
 How many unique bedrooms: 152


In [26]:
print(f"Unique bedroom counts: {np.unique(age)}")
print(f" How many unique bedrooms: {len(np.unique(age))}")

Unique bedroom counts: [29. 34. 35. 37. 38. 39. 40. 41. 42. 43. 44. 45. 46. 47. 48. 49. 50. 51.
 52. 53. 54. 55. 56. 57. 58. 59. 60. 61. 62. 63. 64. 65. 66. 67. 68. 69.
 70. 71. 74. 76. 77.]
 How many unique bedrooms: 41


# correlation

In [28]:
corr_heart_cp=np.corrcoef(age,cp,rowvar=False)
print(f"Corr age and CP {corr_heart_cp}")

Corr age and CP [[ 1.         -0.06865302]
 [-0.06865302  1.        ]]


In [29]:
corr_heart_chol=np.corrcoef(age,chol,rowvar=False)
print(f"Corr age and chol {corr_heart_chol}")

Corr age and chol [[1.         0.21367796]
 [0.21367796 1.        ]]


In [30]:
corr_heart_cp_chol=np.corrcoef(chol,cp,rowvar=False)
print(f"Corr chol and CP {corr_heart_cp_chol}")

Corr chol and CP [[ 1.         -0.07690439]
 [-0.07690439  1.        ]]


* age and chol (0.213) weak and positive
* age and cp (-0.068) weak and negative
* cp and chol (-0.077) weak and negative

# summary

Based on the three key values selected of age, colesterol, and chest pain. there is not a very significant correlation between these values. all three combinations of chol and cp, chol and age, and age and cp result in relativly weak correlations with age and cp and chol and cp having negative relationships, while age and chol has a positive relationship. thus, based on the data presented a relationship between the key values is not certain. however, it is likely that given more data or a different correlation model there could be a statistically significant result. 